src/domain/applications/ApplicationsPageContainer.tsx

```diff
@@ -1,37 +1,101 @@
 import React from 'react';
 import { useTranslation } from 'react-i18next';
+import { useQuery } from '@apollo/react-hooks';
 
 import ApplicationsPage from './ApplicationsPage';
-import Table from '../../common/table/Table';
+import Table, { Column } from '../../common/table/Table';
 import InternalLink from '../../common/internalLink/InternalLink';
+import ApplicationDetails from './applicationDetails/ApplicationDetails';
+import { BERTH_APPLICATIONS_QUERY } from './queries';
+import { BERTH_APPLICATIONS } from './__generated__/BERTH_APPLICATIONS';
+import { getBerthApplicationData, formatDate, ApplicationData } from './utils';
+import LoadingSpinner from '../../common/spinner/LoadingSpinner';
+import Chip from '../../common/chip/Chip';
+
+export interface TableData {
+  id: string;
+  email?: string;
+  phone?: string;
+  address?: string;
+  city?: string;
+  postalCode?: string;
+  berths?: string;
+  boats?: string;
+  group?: string;
+  invoice?: string;
+  name: string;
+  startDate?: string;
+}
+
+type ColumnType = Column<ApplicationData> & { accessor: keyof ApplicationData };
+
+const APPLICATION_STATUS = {
+  PENDING: 'applications.status.pending',
+  OFFER_GENERATED: 'applications.status.offerGenerated',
+  OFFER_SENT: 'applications.status.offerSent',
+  HANDLED: 'applications.status.handled',
+  EXPIRED: 'applications.status.expired',
+  NO_SUITABLE_BERTHS: 'applications.status.noSuitable',
+  NO_SUITABLE_BERTHS_NOTIFIED: 'applications.status.noSuitableNotified',
+};
+
+const APPLICATION_STATUS_COLOR = {
+  PENDING: 'yellow',
+  OFFER_GENERATED: 'blue',
+  OFFER_SENT: 'green',
+  HANDLED: 'blue',
+  EXPIRED: 'grey',
+  NO_SUITABLE_BERTHS: 'red',
+  NO_SUITABLE_BERTHS_NOTIFIED: 'orange',
+};
 
 const ApplicationsPageContainer: React.SFC = () => {
-  const { t } = useTranslation();
+  const { t, i18n } = useTranslation();
+  const { loading, error, data } = useQuery<BERTH_APPLICATIONS>(
+    BERTH_APPLICATIONS_QUERY
+  );
 
-  const columns = [
+  if (error) {
+    return <LoadingSpinner isLoading={loading}>error</LoadingSpinner>;
+  }
+
+  const getApplicationType = (isSwitch: boolean) =>
+    isSwitch
+      ? t('applications.applicationType.switchApplication')
+      : t('applications.applicationType.newApplication');
+
+  const columns: ColumnType[] = [
     {
       Cell: ({ cell }) => (
-        <InternalLink to={`/applications/${cell.value.id}`}>
-          {cell.value.type}
+        <InternalLink to={`/applications/${cell.row.original.id}`}>
+          {getApplicationType(cell.value)}
         </InternalLink>
       ),
-      Header: t('applications.tableHeaders.applicationType'),
-      accessor: 'application',
+      Header: t('applications.tableHeaders.applicationType') || '',
```

Comment:
null coalescing ? It's quite unnecessary cause it t() cant be undefined?

Reply:
For some reason, TS started to complain about `t(..)` return value being `null`

Reply:
Weird, we dont have that. But if they complaint about nullable then no problem here

Reply:
It used to work just fine, after some update it started to complain 🤷‍♂

Reply:
Turns out that one car [shut this `null` value off](https://www.i18next.com/overview/configuration-options#translation-defaults) but it doesn't help with the typings.

Reply:
I bumped into the same issue here. Fixed by reverting yarn.lock: https://github.com/City-of-Helsinki/berth-reservations-admin/pull/100/commits/7917c048f11cb2556d3287d055736322ed9791ce

Reply:
https://github.com/City-of-Helsinki/berth-reservations-admin/pull/106